Add record_service and errors

Dannissimo20 opened the pull request.

Dannissimo20 committed:
Add record_service and errors

fasadar2 commented on src/app/api/records.py:
```diff
@@ -10,6 +11,10 @@
+def get_record_service() -> RecordService:
+    record_repo: RecordRepo = get_record_repo()
+    return RecordService(record_repo)
+
```
Зачем этот метод нужен ? не лучше ли сразу в приватное поле кидать ?

fasadar2 replied:
Вынести методы в отдельный класс

fasadar2 commented on src/app/services/record_service.py:
```diff
@@ -0,0 +1,32 @@
+                else:
+                    if item.start_time.date() >= datetime.now().date():
+                        result.append(item)
+        return result
```
Напиши метод получения в который будешь передовать функцию фильтра, к тому же тебе не нужен такой 3х главый иф можно обойтись двухглавым, и почему убрал фильтрацию с бд на бэк ? У пайтона проблемы с производительностью в иттерационных операциях. Либо оставь этот метод, но через генератор, выглядеть будет гораздо лучше. И да функции фильтрации можно вынести в RecordUtils

fasadar2 commented on src/app/api/records.py:
```diff
@@ -10,6 +11,10 @@
 def get_record_repo() -> RecordRepo:
     return RecordRepo(db_writer)
```
Этот метод тоже не понятно зачем нужен

Dannissimo20 committed:
FEAT(api): Add dependency injector conatainer

Create and integrate DP container in record_router

Dannissimo20 committed:
FIX: rewrite function get_active_records

Move function get_active_records from record_service (python interpretator) to record_repo (database side)

Dannissimo20 committed:
FIX(refactor): prettify my code to PEP8

Dannissimo20 committed:
FEAT: add finish game method
